shared_libraries/orca_shared/reconciliation/shared_reconciliation.py

```diff
@@ -23,10 +34,69 @@ def get_partition_name_from_bucket_name(bucket_name: str):
     """
     Used for interacting with the reconcile_s3_object table.
     Provides a valid partition name given an Orca bucket name.
+    Changes to this function may require a DB migration to recreate partitions.
 
     bucket_name: The name of the Orca bucket in AWS.
     """
     partition_name = "reconcile_s3_object_" + bucket_name.replace("-", "_")
     if not partition_name.replace("_", "").isalnum():
         raise Exception(f"'{partition_name}' is not a valid partition name.")
     return partition_name
+
+
+@shared_db.retry_operational_error()
+def update_job(
+    job_id: int,
+    status: OrcaStatus,
+    now: str,
+    error_message: Optional[str],
+    engine: Engine,
+    logger,
+) -> None:
+    """
+    Updates the status entry for a job.
+
+    Args:
+        job_id: The id of the job to associate info with.
+        status: The status to update the job with.
+        now: Datetime returned by datetime.now(timezone.utc).isoformat()
+        error_message: The error to post to the job, if any.
+        engine: The sqlalchemy engine to use for contacting the database.
+        logger: Logger.
+    """
+    try:
+        logger.debug(f"Creating reconcile records for job {job_id}.")
+        with engine.begin() as connection:
+            end_time = None
+            if status == OrcaStatus.ERROR or status == OrcaStatus.SUCCESS:
+                end_time = now
```

Comment:
Should this throw an error if status is OrcaStatus.ERROR and error_message is None?

Reply:
Since this is not a property of the DB, I would prefer not to include this in the DB translation layer.

Reply:
Come to think of it, why do we even have end_time?